shared/gh/js/controllers/gh.api.util.js

```diff
@@ -139,6 +137,74 @@ define(['exports', 'moment', 'bootstrap-notify'], function(exports, moment) {
     };
 
     /**
+     * Return the current academic week number if the current date is within a term
+     *
+     * @param  {Number}    date     The date where the academic week number needs to be returned for
+     * @return {Number}             The academic week number
+     */
+    var getAcademicWeekNumber = exports.getAcademicWeekNumber = function(date) {
+        if (!_.isNumber(date)) {
+            throw new Error('A valid date should be provided');
+        }
+
+        // Get the configuration
+        var config = require('gh.core').config;
+        // Get the correct terms associated to the current application
+        var terms = config.terms[config.academicYear];
+
+        // Retrieve the corresponding term of the specified date
+        var currentTerm = getTerm(date);
+        if (!currentTerm) {
+            return null;
+        }
+
+        // Get the start date of the corresponding term
+        var startDate = convertISODatetoUnixDate(moment(currentTerm.start).utc().format());
+
+        // Return the current academic week number
+        return Math.ceil((date - startDate) / (PERIODS['week']) - 0.25) + 1;
+    };
+
+    /**
+     * Return the current term if the current date is within a term
+     *
+     * @param  {Number}    date     The date in a UNIX time format
+     * @return {String}             The name of the term that has the specified date in its range
+     */
+    var getTerm = exports.getTerm = function(date) {
+        if (!_.isNumber(date)) {
+            throw new Error('A valid date should be provided');
+        }
+
+        // Get the configuration
+        var config = require('gh.core').config;
+        // Get the correct terms associated to the current application
+        var terms = config.terms[config.academicYear];
+
+        // Get the current term and return its name
+        return _.find(terms, function(term) {
+
+            // Convert the dates from ISO to UNIX for easier calculation
+            var startDate = convertISODatetoUnixDate(moment(term.start).utc().format());
+            var endDate = convertISODatetoUnixDate(moment(term.end).utc().format());
+
+            // We consider weeks that only count 2 days as full academic weeks.
+            // E.g. A term starts on Tuesday 13th February, but the academic weeks always start on a Thursday.
+            //      This means that the first academic week of that term starts on the 13th and ends on the 14th.
+            //      The second academic week will start on Thursday 15th February.
+            var datePlus = convertISODatetoUnixDate(moment(date).add({'days': 6}).utc().format());
+            if (date < startDate && datePlus >= startDate) {
+                date = datePlus;
+            }
+
+            // Return the term where the specified date is within the range
+            if (isDateInRange(date, startDate, endDate)) {
+                return term.name;
```

Comment:
When returning in a `_.find` function it will return the full term object, not just the name.